opentelemetry/src/metrics/instruments/counter.rs

```diff
@@ -25,12 +25,16 @@ where
 impl<T> Counter<T> {
     /// Create a new counter.
     pub fn new(inner: Arc<dyn SyncInstrument<T> + Send + Sync>) -> Self {
-        Counter(inner)
+        Counter(Arc::downgrade(&inner))
     }
 
     /// Records an increment to the counter.
     pub fn add(&self, value: T, attributes: &[KeyValue]) {
-        self.0.measure(value, attributes)
+        if let Some(inner) = self.0.upgrade() {
```

Comment:
how much is the perf cost for this upgrade as opposed to storing Arc itself?

Reply:
There is a considerable perf cost. Check this https://github.com/open-telemetry/opentelemetry-rust/issues/2478#issuecomment-2573491710.